pkg/storage/unified/resource/index.go

```diff
@@ -207,6 +253,17 @@ func (i *Index) Index(ctx context.Context, data *Data) error {
 	}
 	tenant := res.Namespace
 	logger.Debug("indexing resource for tenant", "res", string(data.Value.Value), "tenant", tenant)
+
+	// if tenant doesn't exist, they may have been created during initial indexing
```

Comment:
Since we get all tenants before indexing, a tenant could be created during the indexing process. So if we get an index update and that tenant doesn't have an index, we create and build the index for that tenant.

Reply:
maybe add to the comment